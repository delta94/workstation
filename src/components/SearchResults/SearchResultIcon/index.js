import React from 'react';
import cc from 'classcat';

import { ReactComponent as Bing } from './icons/Bing.svg';
import { ReactComponent as Cisco } from './icons/Cisco.svg';
import { ReactComponent as Confluence } from './icons/Confluence.svg';
import { ReactComponent as Desk } from './icons/Desk.svg';
import { ReactComponent as Google } from './icons/Google.svg';
import { ReactComponent as GoogleDrive } from './icons/GoogleDrive.svg';
import { ReactComponent as Inbenta } from './icons/Inbenta.svg';
import { ReactComponent as KnowledgeOwl } from './icons/KnowledgeOwl.svg';
import { ReactComponent as NoLogo } from './icons/NoLogo.svg';
import { ReactComponent as Okta } from './icons/Okta.svg';
import { ReactComponent as Parature } from './icons/Parature.svg';
import { ReactComponent as Salesforce } from './icons/Salesforce.svg';
import { ReactComponent as YouTube } from './icons/YouTube.svg';
import { ReactComponent as ZenDesk } from './icons/ZenDesk.svg';

import classes from './styles.module.scss';

const iconsByType = {
  default: NoLogo,
  bing: Bing,
  cisco: Cisco,
  confluence: Confluence,
  desk: Desk,
  google: Google,
  googledrive: GoogleDrive,
  inbenta: Inbenta,
  knowledgeowl: KnowledgeOwl,
  okta: Okta,
  parature: Parature,
  salesforce: Salesforce,
  youtube: YouTube,
  zendesk: ZenDesk,
};

export default function SearchResultIcon({ type, className = '' }) {
  const IconComponent = iconsByType[type] ?? iconsByType.default;
  return <IconComponent className={cc([classes['search-result-icon'], className])} />;
}
