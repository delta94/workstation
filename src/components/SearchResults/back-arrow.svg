<svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" viewBox="0 0 17 18">
    <path fill="currentColor" d="M8.7 18l1.2-1.2-6.6-6.4H17V8.6H3.4L10 2.2 8.7 1 0 9.5 8.7 18z"/>
</svg>