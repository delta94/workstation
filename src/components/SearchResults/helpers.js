import walkme from '@walkme/sdk';
import uniqWith from 'lodash/uniqWith';
import isEqual from 'lodash/isEqual';

function findInNode(node, searchTerm) {
  const description = node.description ?? '';
  const url = node.properties.url ?? '';
  const keywords = node.keywords?.join(' ') ?? '';

  return [node.title, description, url, keywords].some((string) =>
    string.toLowerCase().includes(searchTerm.toLowerCase())
  );
}

function flattenUITree(uiTree) {
  const flattenedContent = [];

  uiTree
    .map((tab) => tab.childNodes)
    .flat(1)
    .forEach((node) => {
      node.type === 'category' ? flattenedContent.push(...node.childNodes) : flattenedContent.push(node);
    });

  return uniqWith(flattenedContent, isEqual);
}

export function findInUiTree(searchTerm, uiTree) {
  return (
    flattenUITree(uiTree)
      .filter((node) => ![walkme.content.TypeNames.Survey].includes(node.type))
      // get items that include the search term in their attributes
      .filter((node) => findInNode(node, searchTerm))
  );
}

export async function findInSearchApi(searchTerm, wmSearch) {
  try {
    const searchResults = await wmSearch.search(searchTerm);
    return searchResults.filter((item) => item.type === walkme.content.TypeNames.SearchResult);
  } catch (err) {
    console.error('Could not get search results from the API.', err);
  }
}
