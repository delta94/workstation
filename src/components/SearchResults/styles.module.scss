.search-results {
  list-style-type: none;
  margin: 30px 0;
  padding: 0;

  .return {
    color: var(--neutral500);
    font-style: italic;
    cursor: pointer;
    font-size: 12px;
    display: flex;
    margin-left: 30px;
    margin-bottom: 20px;
    align-items: center;

    &:hover {
      color: var(--primary);
    }

    svg {
      color: var(--primary);
      margin-right: 20px;
    }
  }
}

.search-result-item {
  width: 100%;
  margin-left: 0;
  padding: 17px 30px;
  cursor: pointer;
  display: flex;
  flex-direction: row;

  &:hover {
    .help-list-icon,
    .title {
      color: var(--primary);
    }
  }

  .icon {
    display: inline-block;
    margin-right: 17px;
    color: var(--secondary500);
    flex-shrink: 0;
  }

  .title {
    color: var(--neutral900);
    font-weight: 500;
    font-size: 15px;
    vertical-align: bottom;
  }

  .knowledge-base-type {
    display: inline-block;
    color: var(--secondary500);
    font-weight: 100;
    white-space: nowrap;
    margin-left: 16px;
    font-size: 12px;
  }

  .description {
    font-size: 13px;
    line-height: 1.6;
    color: var(--neutral500);
    padding-top: 4px;
  }

  .url {
    color: var(--secondary500);
    font-weight: 100;
    font-size: 13px;
  }

  &:hover {
    background-color: var(--secondary100);
  }
}
