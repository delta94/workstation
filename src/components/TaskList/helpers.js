function flattenNodes(content) {
  const flattenedContent = [];

  content.forEach((node) => {
    node.type === 'category' ? flattenedContent.push(...node.childNodes) : flattenedContent.push(node);
  });

  return flattenedContent;
}

export function calculateCompletedPercent(content) {
  const flatTasks = flattenNodes(content);
  const total = flatTasks.length;
  const completed = flatTasks.filter((node) => node.properties.isCompleted).length;
  return Math.round((completed / total) * 100);
}
