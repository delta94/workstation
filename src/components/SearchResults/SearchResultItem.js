import React from 'react';

import classes from './styles.module.scss';
import SearchResultIcon from './SearchResultIcon';

export default function SearchResultItem({ node }) {
  function onClick() {
    // ...
  }

  return (
    <li className={classes['search-result-item']} onClick={onClick}>
      <SearchResultIcon type={node.properties.knowledgeBaseType.toLowerCase()} className={classes['icon']} />
      <div>
        <span className={classes.title}>{node.title}</span>
        <span className={classes['knowledge-base-type']}>{node.properties.knowledgeBaseType}</span>
        <div className={classes.description}>{node.description}</div>
        {node.properties?.url && <span className={classes.url}>{node.properties?.url}</span>}
      </div>
    </li>
  );
}
