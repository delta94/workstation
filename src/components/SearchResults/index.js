import React, { useContext, useEffect, useState } from 'react';

import { WalkmeSDKContext } from '../../providers/WalkmeSDKProvider';

import { findInSearchApi, findInUiTree } from './helpers';
import classes from './styles.module.scss';

export default function SearchResults({ searchTerm }) {
  const { uiTreeSDK: uiTree, wmSearch } = useContext(WalkmeSDKContext);
  const [uiTreeResults, setUiTreeResults] = useState(undefined);
  const [apiSearchResults, setApiSearchResults] = useState(undefined);

  useEffect(() => {
    (async function fetchData() {
      setApiSearchResults(await findInSearchApi(searchTerm, wmSearch));
      setUiTreeResults(findInUiTree(searchTerm, uiTree));
    })();
  }, [searchTerm]);

  return (
    <div className={classes['search-results']} style={{ whiteSpace: 'pre' }}>
      <div>{JSON.stringify(uiTreeResults, null, 2)}</div>
      <div>{JSON.stringify(apiSearchResults, null, 2)}</div>
    </div>
  );
}
