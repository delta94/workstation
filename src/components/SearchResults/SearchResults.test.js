import React from 'react';
import { render, act, waitForElement, waitForElementToBeRemoved } from '@testing-library/react';

import { sdkProvider } from '../../helpers/sdkMockForTests';
import { WalkmeSDKContext } from '../../providers/WalkmeSDKProvider';

import SearchResults from './index';

jest.mock('@walkme/sdk', () => ({
  content: {
    TypeNames: {
      Video: 'video',
      Article: 'article',
      Launcher: 'launcher',
      ShoutOut: 'shoutOut',
      Shuttle: 'shuttle',
      SmartWalkThru: 'smart-walkthru',
      Survey: 'survey',
      Task: 'task',
      Course: 'course',
      Lesson: 'lesson',
      Walkthru: 'walkthru',
      Category: 'category',
      Tab: 'tab',
      Experiments: 'experiments',
      SearchProviderUrl: 'search-provider-url',
      Tag: 'tag',
      TrackedElement: 'trackedElement',
      TrackedPage: 'trackedPage',
      SupportItem: 'support',
      SearchResult: 'search-result',
      HelpDesk: 'helpDesk',
      SmartTipSet: 'smartTipSet',
    },
  },
}));

describe('SearchResults renders correctly', () => {
  afterAll(() => {
    jest.restoreAllMocks();
  });

  test('renders correctly', async () => {
    await act(async () => {
      const searchTerm = 'content';

      const { container, getByText, getByTestId } = render(
        <WalkmeSDKContext.Provider value={sdkProvider}>
          <SearchResults searchTerm={searchTerm} />
        </WalkmeSDKContext.Provider>
      );

      await waitForElementToBeRemoved(() => getByText('Loading'));
      await waitForElement(() => getByTestId('search-results'));
      expect(container).toMatchSnapshot();
    });
  });
});
