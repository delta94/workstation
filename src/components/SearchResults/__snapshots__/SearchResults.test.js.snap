// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults renders correctly renders correctly 1`] = `
<div>
  <div
    class="search-results"
    data-testid="search-results"
  >
    <button
      class="return"
    >
      <svg>
        back-arrow.svg
      </svg>
      Results for "
      content
      "
    </button>
    <ul
      class="search-results-list"
    >
      <li
        class="help-item"
      >
        <div
          class="help-list-icon"
        >
          <svg
            class="help-icon"
          >
            Walkthru.svg
          </svg>
        </div>
        <div
          class="help-item-text"
        >
          <div
            class="title"
          >
            Content smart-walkthru-280
          </div>
          <div
            class="description"
          >
            Description for smart-walkthru-280
          </div>
        </div>
      </li>
      <li
        class="help-item"
      >
        <div
          class="help-list-icon"
        >
          <svg
            class="help-icon"
          >
            Shuttle.svg
          </svg>
        </div>
        <div
          class="help-item-text"
        >
          <div
            class="title"
          >
            Content shuttle-27
          </div>
          <div
            class="description"
          >
            Description for shuttle-27
          </div>
          <div
            class="url"
          >
            https://27.example.com/index?t=599
          </div>
        </div>
      </li>
      <li
        class="help-item"
      >
        <div
          class="help-list-icon"
        >
          <svg
            class="help-icon"
          >
            Video.svg
          </svg>
        </div>
        <div
          class="help-item-text"
        >
          <div
            class="title"
          >
            Content video-981
          </div>
          <div
            class="description"
          >
            Description for video-981
          </div>
        </div>
      </li>
      <li
        class="help-item"
      >
        <div
          class="help-list-icon"
        >
          <svg
            class="help-icon"
          >
            Article.svg
          </svg>
        </div>
        <div
          class="help-item-text"
        >
          <div
            class="title"
          >
            Content article-723
          </div>
          <div
            class="description"
          >
            Description for article-723
          </div>
        </div>
      </li>
      <li
        class="help-item"
      >
        <div
          class="help-list-icon"
        >
          <svg
            class="help-icon"
          >
            Walkthru.svg
          </svg>
        </div>
        <div
          class="help-item-text"
        >
          <div
            class="title"
          >
            Content smart-walkthru-554
          </div>
          <div
            class="description"
          >
            Description for smart-walkthru-554
          </div>
        </div>
      </li>
      <li
        class="help-item"
      >
        <div
          class="help-list-icon"
        >
          <svg
            class="help-icon"
          >
            Shuttle.svg
          </svg>
        </div>
        <div
          class="help-item-text"
        >
          <div
            class="title"
          >
            Content shuttle-70
          </div>
          <div
            class="description"
          >
            Description for shuttle-70
          </div>
          <div
            class="url"
          >
            https://27.example.com/index?t=599
          </div>
        </div>
      </li>
      <li
        class="help-item"
      >
        <div
          class="help-list-icon"
        >
          <svg
            class="help-icon"
          >
            Video.svg
          </svg>
        </div>
        <div
          class="help-item-text"
        >
          <div
            class="title"
          >
            Content video-896
          </div>
          <div
            class="description"
          >
            Description for video-896
          </div>
        </div>
      </li>
      <li
        class="help-item"
      >
        <div
          class="help-list-icon"
        >
          <svg
            class="help-icon"
          >
            Article.svg
          </svg>
        </div>
        <div
          class="help-item-text"
        >
          <div
            class="title"
          >
            Content article-522
          </div>
          <div
            class="description"
          >
            Description for article-522
          </div>
        </div>
      </li>
      <li
        class="help-item"
      >
        <div
          class="help-list-icon"
        >
          <svg
            class="help-icon"
          >
            Shuttle.svg
          </svg>
        </div>
        <div
          class="help-item-text"
        >
          <div
            class="title"
          >
            Content shuttle-518
          </div>
          <div
            class="description"
          >
            Description for shuttle-518
          </div>
          <div
            class="url"
          >
            https://27.example.com/index?t=599
          </div>
        </div>
      </li>
      <li
        class="help-item"
      >
        <div
          class="help-list-icon"
        >
          <svg
            class="help-icon"
          >
            Shuttle.svg
          </svg>
        </div>
        <div
          class="help-item-text"
        >
          <div
            class="title"
          >
            Content shuttle-492
          </div>
          <div
            class="description"
          >
            Description for shuttle-492
          </div>
          <div
            class="url"
          >
            https://27.example.com/index?t=599
          </div>
        </div>
      </li>
      <li
        class="help-item"
      >
        <div
          class="help-list-icon"
        >
          <svg
            class="help-icon"
          >
            Shuttle.svg
          </svg>
        </div>
        <div
          class="help-item-text"
        >
          <div
            class="title"
          >
            Content shuttle-971
          </div>
          <div
            class="description"
          >
            Description for shuttle-971
          </div>
          <div
            class="url"
          >
            https://27.example.com/index?t=599
          </div>
        </div>
      </li>
      <li
        class="help-item"
      >
        <div
          class="help-list-icon"
        >
          <svg
            class="help-icon"
          >
            Shuttle.svg
          </svg>
        </div>
        <div
          class="help-item-text"
        >
          <div
            class="title"
          >
            Content shuttle-708
          </div>
          <div
            class="description"
          >
            Description for shuttle-708
          </div>
          <div
            class="url"
          >
            https://27.example.com/index?t=599
          </div>
        </div>
      </li>
      <li
        class="help-item"
      >
        <div
          class="help-list-icon"
        >
          <svg
            class="help-icon"
          >
            Shuttle.svg
          </svg>
        </div>
        <div
          class="help-item-text"
        >
          <div
            class="title"
          >
            Content shuttle-132
          </div>
          <div
            class="description"
          >
            Description for shuttle-132
          </div>
          <div
            class="url"
          >
            https://27.example.com/index?t=599
          </div>
        </div>
      </li>
      <li
        class="help-item"
      >
        <div
          class="help-list-icon"
        >
          <svg
            class="help-icon"
          >
            Shuttle.svg
          </svg>
        </div>
        <div
          class="help-item-text"
        >
          <div
            class="title"
          >
            Content shuttle-381
          </div>
          <div
            class="description"
          >
            Description for shuttle-381
          </div>
          <div
            class="url"
          >
            https://27.example.com/index?t=599
          </div>
        </div>
      </li>
      <li
        class="help-item"
      >
        <div
          class="help-list-icon"
        >
          <svg
            class="help-icon"
          >
            Shuttle.svg
          </svg>
        </div>
        <div
          class="help-item-text"
        >
          <div
            class="title"
          >
            Content shuttle-304
          </div>
          <div
            class="description"
          >
            Description for shuttle-304
          </div>
          <div
            class="url"
          >
            https://27.example.com/index?t=599
          </div>
        </div>
      </li>
      <li
        class="help-item"
      >
        <div
          class="help-list-icon"
        >
          <svg
            class="help-icon"
          >
            Shuttle.svg
          </svg>
        </div>
        <div
          class="help-item-text"
        >
          <div
            class="title"
          >
            Content shuttle-353
          </div>
          <div
            class="description"
          >
            Description for shuttle-353
          </div>
          <div
            class="url"
          >
            https://27.example.com/index?t=599
          </div>
        </div>
      </li>
      <li
        class="help-item"
      >
        <div
          class="help-list-icon"
        >
          <svg
            class="help-icon"
          >
            Shuttle.svg
          </svg>
        </div>
        <div
          class="help-item-text"
        >
          <div
            class="title"
          >
            Content shuttle-548
          </div>
          <div
            class="description"
          >
            Description for shuttle-548
          </div>
          <div
            class="url"
          >
            https://27.example.com/index?t=599
          </div>
        </div>
      </li>
      <li
        class="help-item"
      >
        <div
          class="help-list-icon"
        >
          <svg
            class="help-icon"
          >
            Shuttle.svg
          </svg>
        </div>
        <div
          class="help-item-text"
        >
          <div
            class="title"
          >
            Content shuttle-481
          </div>
          <div
            class="description"
          >
            Description for shuttle-481
          </div>
          <div
            class="url"
          >
            https://27.example.com/index?t=599
          </div>
        </div>
      </li>
      <li
        class="search-result-item"
      >
        <div
          class="icon"
        >
          <svg
            class="search-result-icon"
          >
            Google.svg
          </svg>
        </div>
        <div
          class="search-result-text"
        >
          <div
            class="title"
          >
            Peace to Prosperity　
          </div>
          <div
            class="knowledge-base-type"
          >
            Google
          </div>
          <div
            class="description"
          >
            Jan 1, 2020 ... CONTENTS. GLOSSARY | 1 ... adjustments to the share of Egyptian, Israeli, or 
Palestinian content required to meet the 35 percent content.
          </div>
          <div
            class="url"
          >
            http://pic-upload.ynet.co.il/news/Peace-to-Prosperity-0120.pdf
          </div>
        </div>
      </li>
      <li
        class="search-result-item"
      >
        <div
          class="icon"
        >
          <svg
            class="search-result-icon"
          >
            Google.svg
          </svg>
        </div>
        <div
          class="search-result-text"
        >
          <div
            class="title"
          >
            USDA National Fluoride Database of Selected Beverages and...　
          </div>
          <div
            class="knowledge-base-type"
          >
            Google
          </div>
          <div
            class="description"
          >
            Fluoride content of municipal water in the United States: What percentage is 
fluoridated? J. Food Comp. Anal. 16(5): 621-628. Ophaug RH. 1983-1987.
          </div>
          <div
            class="url"
          >
            http://go.ynet.co.il/pic/briut/F02.pdf
          </div>
        </div>
      </li>
      <li
        class="search-result-item"
      >
        <div
          class="icon"
        >
          <svg
            class="search-result-icon"
          >
            Google.svg
          </svg>
        </div>
        <div
          class="search-result-text"
        >
          <div
            class="title"
          >
            Fat and Energy Contents of Expressed Human Breast Milk in...　
          </div>
          <div
            class="knowledge-base-type"
          >
            Google
          </div>
          <div
            class="description"
          >
            Energy contents of the milk were measured with a bomb calorimeter. Results. 
The groups did not differ in terms of mater- nal height and diet, infant birth weight,
 ...
          </div>
          <div
            class="url"
          >
            http://prdupl02.ynet.co.il/ForumFiles/13424549.pdf
          </div>
        </div>
      </li>
      <li
        class="search-result-item"
      >
        <div
          class="icon"
        >
          <svg
            class="search-result-icon"
          >
            Google.svg
          </svg>
        </div>
        <div
          class="search-result-text"
        >
          <div
            class="title"
          >
            אמבלייז רוכשת השליטה ב-Smart Content מרעננה　
          </div>
          <div
            class="knowledge-base-type"
          >
            Google
          </div>
          <div
            class="description"
          >
            16 ינואר 2006 ... Smart Content מפתח טכנולוגיה ל"דחיפת" תוכן מותאם אישית למכשירים ניידים ומחשבים 
אישיים על גבי רשתות אלחוטיות ו-IP. אמבלייז מתכוונת לשלב את טכנולוגיית ...
          </div>
          <div
            class="url"
          >
            https://www.ynet.co.il/articles/0,7340,L-3201198,00.html
          </div>
        </div>
      </li>
      <li
        class="search-result-item"
      >
        <div
          class="icon"
        >
          <svg
            class="search-result-icon"
          >
            Google.svg
          </svg>
        </div>
        <div
          class="search-result-text"
        >
          <div
            class="title"
          >
            מפת הבחירות של ישראל 2019 - ynet　
          </div>
          <div
            class="knowledge-base-type"
          >
            Google
          </div>
          <div
            class="description"
          >
            המידע המוצג מבוסס על עיבוד שערך אתר ״מדלן - הכל על בתים ושכונות״ לנתוני ועדת הבחירות 
המרכזית. נתוני השכונות מקורבים לפי התאמת קלפיות. כל המידע מוצג תחת תנאי השימוש ...
          </div>
          <div
            class="url"
          >
            https://z.ynet.co.il/short/content/ElectionMap2019/
          </div>
        </div>
      </li>
      <li
        class="search-result-item"
      >
        <div
          class="icon"
        >
          <svg
            class="search-result-icon"
          >
            Google.svg
          </svg>
        </div>
        <div
          class="search-result-text"
        >
          <div
            class="title"
          >
            X-rays　
          </div>
          <div
            class="knowledge-base-type"
          >
            Google
          </div>
          <div
            class="description"
          >
            hirty years ago, X-rays were the only way to see what was going on inside your 
body. Now other methods of medical imaging are available, some using different
 ...
          </div>
          <div
            class="url"
          >
            http://prdupl02.ynet.co.il/ForumFiles_2/16567358.pdf
          </div>
        </div>
      </li>
      <li
        class="search-result-item"
      >
        <div
          class="icon"
        >
          <svg
            class="search-result-icon"
          >
            Google.svg
          </svg>
        </div>
        <div
          class="search-result-text"
        >
          <div
            class="title"
          >
            Vizual.ai　
          </div>
          <div
            class="knowledge-base-type"
          >
            Google
          </div>
          <div
            class="description"
          >
            Content Publishers, Photo Editors & Online Retailers can now leverage the 
power of the Vizual.ai platform to select the optimal imagery for their customers.
          </div>
          <div
            class="url"
          >
            https://yq-api.ynet.co.il/
          </div>
        </div>
      </li>
      <li
        class="search-result-item"
      >
        <div
          class="icon"
        >
          <svg
            class="search-result-icon"
          >
            Google.svg
          </svg>
        </div>
        <div
          class="search-result-text"
        >
          <div
            class="title"
          >
            גוגל: בטא של תוכנית Adsense for Content בעברית　
          </div>
          <div
            class="knowledge-base-type"
          >
            Google
          </div>
          <div
            class="description"
          >
            27 דצמבר 2006 ... השירות של גוגל, המאפשר הצבת מודעות קשורות תוכן, הושק בעברית בבטא, בשלב ראשון רק 
באתר תפוז.
          </div>
          <div
            class="url"
          >
            https://www.ynet.co.il/articles/0,7340,L-3345160,00.html
          </div>
        </div>
      </li>
      <li
        class="search-result-item"
      >
        <div
          class="icon"
        >
          <svg
            class="search-result-icon"
          >
            Google.svg
          </svg>
        </div>
        <div
          class="search-result-text"
        >
          <div
            class="title"
          >
            (function() { CmmAppVideoApi.DecorationTypes = { Credit: ...　
          </div>
          <div
            class="knowledge-base-type"
          >
            Google
          </div>
          <div
            class="description"
          >
            _registerContentEvents(content); this.pool.push(content); }; }; this.
_registerContentEvents = function(content){ var contentEvents = content.
getEventList(); ...
          </div>
          <div
            class="url"
          >
            https://www.ynet.co.il/Cmn/App/Video/CmmAppVideoApi_Js_Content/0,13943,280716135620,00.html
          </div>
        </div>
      </li>
      <li
        class="search-result-item"
      >
        <div
          class="icon"
        >
          <svg
            class="search-result-icon"
          >
            Google.svg
          </svg>
        </div>
        <div
          class="search-result-text"
        >
          <div
            class="title"
          >
            תחרות בתחומי העיצוב וה-e-Content　
          </div>
          <div
            class="knowledge-base-type"
          >
            Google
          </div>
          <div
            class="description"
          >
            23 יוני 2006 ... Top Talent של EUROPRIX היא תחרות אירופית עבור סטודנטים ואנשי מקצוע צעירים, 
העוסקים בפרויקטים חדשניים בתחומי העיצוב וה- e-Content (תוכן אלקטרוני), ...
          </div>
          <div
            class="url"
          >
            https://www.ynet.co.il/articles/0,7340,L-3266501,00.html
          </div>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
